core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -528,7 +528,13 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
         new StageUIData
       })
       val taskData = stageData.taskData.get(taskId)
-      val metrics = TaskMetrics.fromAccumulatorInfos(accumUpdates)
+      val accumsFiltered = if (conf.get(TASK_METRICS_TRACK_UPDATED_BLOCK_STATUSES)) {
+        accumUpdates
+      } else {
+        accumUpdates.filter(info => info.name.isDefined && info.update.isDefined && info.name !=
```

Comment:
I'm not sure what you mean by its already stored?  It gets stored into the TaskMetrics when the call below to TaskMetrics.fromAccumulatorInfo is made, now the task metrics UpdatedBlockStatuses it returns aren't really ever used by this function in updateAggregateMetrics or updateTaskMetric., but I didn't see any reason to set it since its not used.